lib/rules/prefer-regex-literals.js

```diff
@@ -151,6 +243,55 @@ module.exports = {
             return false;
         }
 
+        /* eslint-disable jsdoc/valid-types -- eslint-plugin-jsdoc's type parser doesn't support square brackets */
+        /**
+         * Returns a ecmaVersion compatible for regexpp.
+         * @param {import("../linter/linter").ParserOptions["ecmaVersion"]} ecmaVersion The ecmaVersion to convert.
+         * @returns {import("regexpp/ecma-versions").EcmaVersion} The resulting ecmaVersion compatible for regexpp.
+         */
+        function getRegexppEcmaVersion(ecmaVersion) {
+            /* eslint-enable jsdoc/valid-types -- JSDoc is over, enable jsdoc/valid-types again */
```

Comment:
```suggestion
        /**
         * Returns a ecmaVersion compatible for regexpp.
         * @param {any} ecmaVersion The ecmaVersion to convert.
         * @returns {import("regexpp/ecma-versions").EcmaVersion} The resulting ecmaVersion compatible for regexpp.
         */
        function getRegexppEcmaVersion(ecmaVersion) {
```

Type for `ecmaVersion` should be `any`. It seems that we can also remove `eslint-disable jsdoc/valid-types` after this change.